Description: When overwriting existing symlinks, treat ENOKEY like ENOENT.
 When trying to overwrite existing symlinks, dpkg tries to stat the target of
 the symlink to check if it points to a directory. Treate ENOKEY like ENOENT,
 as the point of the check is to ensure that the new file doesn't end up in the
 directory the symlink could be pointing to.
Author: Sven Mueller <sven.mueller72@gmail.com>
Last-Update: 2020-08-26
---
Index: dpkg-1.19.7/src/archives.c
===================================================================
--- dpkg-1.19.7.orig/src/archives.c
+++ dpkg-1.19.7/src/archives.c
@@ -626,7 +626,8 @@ linktosameexistingdir(const struct tar_e
 
   statr= stat(fname, &oldstab);
   if (statr) {
-    if (!(errno == ENOENT || errno == ELOOP || errno == ENOTDIR))
+    if (!(errno == ENOENT || errno == ELOOP || errno == ENOTDIR ||
+          errno == ENOKEY))
       ohshite(_("failed to stat (dereference) existing symlink '%.250s'"),
               fname);
     return false;
